pkg/component/auditlog-forwarder/deploy_test.go

```diff
@@ -0,0 +1,727 @@
+
+				managedResourceSecret.Name = managedResource.Spec.SecretRefs[0].Name
+				Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(managedResourceSecret), managedResourceSecret)).To(Succeed())
+
+				tlsSecret, found := fakeSecretManager.Get("auditlog-forwarder-tls")
+				Expect(found).To(BeTrue())
+
+				caBundle, found := fakeSecretManager.Get("ca-extension-auditing")
+				Expect(found).To(BeTrue())
+
+				clientTLSSecret, found := fakeSecretManager.Get("auditlog-forwarder-client-tls")
+				Expect(found).To(BeTrue())
+
+				kubeConfig := &clientcmdv1.Config{
+					Clusters: []clientcmdv1.NamedCluster{{
+						Name: "auditing",
+						Cluster: clientcmdv1.Cluster{
+							Server:                   fmt.Sprintf("https://%s.%s.svc:10443/audit", "auditlog-forwarder", namespace),
+							CertificateAuthorityData: caBundle.Data["bundle.crt"],
+							InsecureSkipTLSVerify:    false,
+						},
+					}},
+					Contexts: []clientcmdv1.NamedContext{{
+						Name: "auditing",
+						Context: clientcmdv1.Context{
+							Cluster:  "auditing",
+							AuthInfo: "auditing",
+						},
+					}},
+					CurrentContext: "auditing",
+					AuthInfos: []clientcmdv1.NamedAuthInfo{{
+						Name: "auditing",
+						AuthInfo: clientcmdv1.AuthInfo{
+							ClientCertificateData: clientTLSSecret.Data["tls.crt"],
+							ClientKeyData:         clientTLSSecret.Data["tls.key"],
+						},
+					}},
+				}
+				kubeAPIServerKubeConfig, err := runtime.Encode(clientcmdlatest.Codec, kubeConfig)
+				Expect(err).NotTo(HaveOccurred())
+				kubeconfigSecret.Data = map[string][]byte{
+					"kubeconfig": kubeAPIServerKubeConfig,
+				}
+
+				deployment.Spec.Template.Spec.Volumes[1].Secret.SecretName = tlsSecret.Name
+				deployment.Spec.Template.Spec.Volumes[2].Secret.SecretName = caBundle.Name
+				utilruntime.Must(references.InjectAnnotations(deployment))
+
+				expectedObjects = []client.Object{
+					vpa,
+					serviceAccount,
+					podDisruptionBudget,
+					kubeconfigSecret,
+					service,
+					configMap,
+					deployment,
+				}
+
+				Expect(managedResourceSecret.Type).To(Equal(corev1.SecretTypeOpaque))
+				Expect(managedResourceSecret.Immutable).To(Equal(ptr.To(true)))
+				Expect(managedResourceSecret.Labels["resources.gardener.cloud/garbage-collectable-reference"]).To(Equal("true"))
+			})
+
+			It("should successfully deploy all resources", func() {
+				Expect(managedResource).To(consistOf(expectedObjects...))
+			})
+		})
+	})
+
+	Describe("#Destroy", func() {
+		It("should successfully destroy all resources", func() {
+			Expect(fakeClient.Create(ctx, managedResource)).To(Succeed())
+			Expect(fakeClient.Create(ctx, managedResourceSecret)).To(Succeed())
+
+			Expect(deployer.Destroy(ctx)).To(Succeed())
+
+			Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(BeNotFoundError())
+			Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(managedResourceSecret), managedResourceSecret)).To(BeNotFoundError())
```

Comment:
I think you should assert the existing resources(deployment, secrets, vpa, etc) are deleted, no?

Reply:
This is up to the gardener resource manager. Other similar tests only assert the managed resources.

Reply:
True 😐 ,  I do not know why I was expecting `Destroy` to take care to delete the "managed resource".